homeassistant/components/demo/remote.py

```diff
@@ -21,50 +39,32 @@ def setup_platform(hass, config, add_entities_callback, discovery_info=None):
 class DemoRemote(RemoteEntity):
     """Representation of a demo remote."""
 
-    def __init__(self, name, state, icon):
+    _attr_should_poll = False
+
+    def __init__(self, name: str | None, state: bool, icon: str | None) -> None:
         """Initialize the Demo Remote."""
-        self._name = name or DEVICE_DEFAULT_NAME
-        self._state = state
-        self._icon = icon
+        self._attr_name = name or DEVICE_DEFAULT_NAME
+        self._attr_is_on = state
+        self._attr_icon = icon
         self._last_command_sent = None
 
     @property
-    def should_poll(self):
-        """No polling needed for a demo remote."""
-        return False
-
-    @property
-    def name(self):
-        """Return the name of the device if any."""
-        return self._name
-
-    @property
-    def icon(self):
-        """Return the icon to use for device if any."""
-        return self._icon
-
-    @property
-    def is_on(self):
-        """Return true if remote is on."""
-        return self._state
-
-    @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, Any] | None:
         """Return device state attributes."""
         if self._last_command_sent is not None:
             return {"last_command_sent": self._last_command_sent}
```

Comment:
Side note: We should add an explicit `return None` at the end.

Reply:
#52390